src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,42 @@
+package duke;
+
+
+/**
+ * This class represents a running instance of Duke, the task list!
+ */
+public class Duke {
+
+    private Storage manager;
+
+    private Parser parser;
+
+    private TaskList history;
+
+    private UI ui;
+
+    /**
+     * Constructor for duke application.
+     */
+    public Duke() {
+        this.manager = new Storage("data");
+        this.manager.createLog();
+        this.parser = new Parser();
+        this.history = new TaskList(this.manager);
+        this.ui = new UI(manager, parser, history);
+    }
+
+    /**
+     * Getter for UI.
+     * @return UI for the chatbot.
+     */
+    public UI getUI() {
```

Comment:
Perhaps we could write our method to more closely follow the coding standards!
```suggestion
    public UI getUi() {
```